examples/generative/molecule_generation.py

```diff
@@ -0,0 +1,575 @@
+    # Symmetrify tensors in the last two dimensions
+    x_adjacency = (x_adjacency + tf.transpose(x_adjacency, (0, 1, 3, 2))) / 2
+    x_adjacency = keras.layers.Softmax(axis=1)(x_adjacency)
+
+    # Map outputs of previous layer (x) to [continuous] feature tensors (x_features)
+    x_features = keras.layers.Dense(tf.math.reduce_prod(feature_shape))(x)
+    x_features = keras.layers.Reshape(feature_shape)(x_features)
+    x_features = keras.layers.Softmax(axis=2)(x_features)
+
+    decoder = keras.Model(
+        latent_inputs, outputs=[x_adjacency, x_features], name="decoder"
+    )
+
+    return decoder
+
+
+"""
+## Build Sampling layer
+"""
+
+
+class Sampling(layers.Layer):
+    def call(self, inputs):
+        z_mean, z_log_var = inputs
+        batch = tf.shape(z_log_var)[0]
+        dim = tf.shape(z_log_var)[1]
+        epsilon = tf.keras.backend.random_normal(shape=(batch, dim))
+        return z_mean + tf.exp(0.5 * z_log_var) * epsilon
+
+
+"""
+## Building the VAE
+With this model we try to optimize four losses:
+
+
+* Categorical Crossentropy Loss
+* KL Loss
+* Property prediction Loss
+* Graph Loss( Gredient Penalty )
+
+The Categorical Crossentropy loss is to give a measure of the model's reconstruction
+capacity. The Property prediction loss is to estimate the mean squared error of the
+predicted and actual property after passing the ```z_mean(generated from the encoder)```
+through a property prediction model or property prediction layer. The property prediction
+of model is optimized through binary crossentropy and gradient penalty is further guided
+by the model's property(QED) prediction.
+"""
+
+
+class MoleculeGenerator(keras.Model):
+    def __init__(self, encoder, decoder, max_len, **kwargs):
+        super().__init__(**kwargs)
+        self.encoder = encoder
+        self.decoder = decoder
+        self.property_prediction_layer = layers.Dense(1)
+        self.max_len = max_len
+
+        self.train_total_loss_tracker = keras.metrics.Mean(name="train_total_loss")
+        self.val_total_loss_tracker = keras.metrics.Mean(name="val_total_loss")
+
+    def train_step(self, data):
+        adjacency_tensor, feature_tensor, qed_tensor = data[0]
+        graph_real = [adjacency_tensor, feature_tensor]
+        self.batch_size = tf.shape(qed_tensor)[0]
+        with tf.GradientTape() as tape:
+            z_mean, z_log_var, qed_pred, gen_adjacency, gen_features = self(
+                graph_real, training=True
+            )
+            graph_generated = [gen_adjacency, gen_features]
+            total_loss = self.calculate_loss(
+                z_log_var, z_mean, qed_tensor, qed_pred, graph_real, graph_generated
+            )
+
+        grads = tape.gradient(total_loss, self.trainable_weights)
+        self.optimizer.apply_gradients(zip(grads, self.trainable_weights))
+
+        self.train_total_loss_tracker.update_state(total_loss)
+        return {"loss": self.train_total_loss_tracker.result()}
+
+    def calculate_loss(
```

Comment:
There is a built-in `compute_loss()` method, please use a slightly difference name to avoid confusion. `_calculate_loss()` maybe?